api/proto/v1/addons-service.proto

```diff
@@ -0,0 +1,330 @@
+syntax = "proto3";
+package v1;
+
+import "google/api/annotations.proto";
+import "google/protobuf/any.proto";
+
+option go_package = "github.com/mt-sre/addon-service/pkg/api/v1";
+
+enum AddonParameterValueType {
+    string = 0;
+    boolean = 1;
+    number = 2;
+    cidr = 3;
+    resource = 4;
+    resource_required = 5;
+}
```

Comment:
Enums are a bit tricky.
We should define the 0 key as `unknown` to be forward compatible, when adding new values to the APIs a client that is newer than the API might otherwise run into undefined behavior.

Because a key "6" or "<null>" would be mapped to 0, if the server does not know about it yet.